fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RecordConstructorValue.java

```diff
@@ -377,14 +379,43 @@ public Object finish() {
             }
 
             @Nonnull
-            private List<Accumulator> buildAccumulators() {
+            private List<Accumulator> buildAccumulators(@Nullable List<RecordCursorProto.AccumulatorState> initialState) {
+                List<Value> childrenAsList = new ArrayList<>();
+                getChildren().forEach(childrenAsList::add);
                 final ImmutableList.Builder<Accumulator> childAccumulatorsBuilder = ImmutableList.builder();
-                for (final var child : getChildren()) {
-                    Verify.verify(child instanceof AggregateValue);
-                    childAccumulatorsBuilder.add(((AggregateValue)child).createAccumulator(typeRepository));
+
+                if (initialState != null) {
+                    Verify.verify(initialState.size() == childrenAsList.size());
+                }
+
+                for (int i = 0; i < childrenAsList.size(); i++) {
+                    Verify.verify(childrenAsList.get(i) instanceof AggregateValue);
+                    if (initialState == null) {
+                        childAccumulatorsBuilder.add(((AggregateValue)childrenAsList.get(i)).createAccumulatorWithInitialState(typeRepository, null));
+                    } else {
+                        childAccumulatorsBuilder.add(((AggregateValue)childrenAsList.get(i)).createAccumulatorWithInitialState(typeRepository, List.of(initialState.get(i))));
+                    }
                 }
                 return childAccumulatorsBuilder.build();
             }
+
+            @Nullable
+            @Override
+            public RecordCursorProto.PartialAggregationResult getPartialAggregationResult(Message groupingKey) {
+                List<RecordCursorProto.AccumulatorState> accumulatorStates = new ArrayList<>();
+                for (Accumulator accumulator: childAccumulators) {
+                    if (accumulator.getPartialAggregationResult(groupingKey) != null) {
+                        accumulatorStates.addAll(accumulator.getPartialAggregationResult(groupingKey).getAccumulatorStatesList());
```

Comment:
Updating this method so that it only returns the list of `AccumulatorState`s would seem to tidy this up a bit, in that it's currently attaching the grouping key to each state only to throw it away and then attach it at a higher level

Reply:
modified.